net2/FlowTool.js

```diff
@@ -267,28 +267,17 @@ class FlowTool {
       f.device = flow.mac;
     }
 
-    if(flow.pf) {
+    f.ptocotol = flow.pr;
+
+    try {
       if(flow.lh === flow.sh) {
-        try {
-          const protocol = Object.keys(flow.pf)[0].split(".")[0];
```

Comment:
the code logic is deleted??

Reply:
https://github.com/firewalla/firewalla/blob/38f012f9cec31abaee04075435a18dd494e69c83/net2/BroDetect.js#L883-L884
no need to do string parsing here, flow.pr, flow.sp, flow.dp have all things we need